Sources/AppCoinsSDK/UI/Purchase/ViewModels/BottomSheetViewModel.swift

```diff
@@ -44,7 +45,35 @@ internal class BottomSheetViewModel : ObservableObject {
     internal var walletApplicationUseCases: WalletApplicationUseCases = WalletApplicationUseCases.shared
     internal var currencyUseCases: CurrencyUseCases = CurrencyUseCases.shared
     
-    private init() { UserDefaults.standard.set(false, forKey: "_UIConstraintBasedLayoutLogUnsatisfiable") } // Prevents Layout Warning Prints
+    // Device Orientation
+    @Published var isLandscape: Bool = false
+    
+    // Keyboard Dismiss
+    @Published var isKeyboardVisible: Bool = false
+    private var cancellables = Set<AnyCancellable>()
+    
+    @Published var isCreditCardView: Bool = false
+    
+    private init(isLandscape: Bool) {
+        // Prevents Layout Warning Prints
+        UserDefaults.standard.set(false, forKey: "_UIConstraintBasedLayoutLogUnsatisfiable")
+        
+        NotificationCenter.default.publisher(for: UIResponder.keyboardWillShowNotification)
+            .sink { [weak self] _ in
+                self?.isKeyboardVisible = true
```

Comment:
When we are showing and dismissing the keyboard it would be nice if it could be more smooth, try to implement an animation here or where the changing of this state triggers a view change.